x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -698,6 +703,20 @@ protected void nodeOperation(AllocatedPersistentTask task, StartDataFrameAnalyti
                 return;
             }
 
+            ActionListener<Boolean> templateCheckListener = ActionListener.wrap(
+                ok -> executeTask(analyticsTaskState, task),
+                error -> {
+                    Throwable cause = ExceptionsHelper.unwrapCause(error);
+                    String msg = "Failed to create internal index template [" + inferenceIndexTemplate.getTemplateName() + "]";
+                    logger.error(msg, cause);
+                    task.markAsFailed(error);
```

Comment:
Is this sufficient to mark the task as failed? Is there any other clean up to do?

Reply:
```
task.updatePersistentTaskState(startedState, ActionListener.wrap(
                    response -> manager.execute((DataFrameAnalyticsTask) task, DataFrameAnalyticsState.STARTED, clusterState),
                    task::markAsFailed));
```
only calls `task.markAsFailed`, so this should be adequate. 

@dimitris-athanasiou do you have a different opinion?

Reply:
Yes, it should be enough